src/reprs/binary32.rkt

```diff
@@ -87,7 +87,7 @@
   (shift 31 ordinal->float32)
   (unshift 31 float32->ordinal)
   32
-  (disjoin nan? infinite?))
+  (λ (x) (and (number? x) (or (nan? x) (infinite? x)))))
```

Comment:
Huh, does this ever make a difference? If so, good catch. Hopefully we'll remove the `infinite?` bit one day and then it can go back to `(conjoin number? infinite?)`.